scripts/testramfs/testramfs.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/u-root/u-root/pkg/cpio"
+	_ "github.com/u-root/u-root/pkg/cpio/newc"
+)
+
+func main() {
+	flag.Parse()
+
+	if flag.NArg() != 1 {
+		log.Fatalln("usage: %s <cpio-path>", os.Args[0])
+	}
+
+	if err := Test(flag.Args()[0]); err != nil {
+		log.Fatalln(err)
+	}
+}
+
+func Test(name string) error {
+	// So, what's the plan here?
+	//
+	// - new mount namespace
+	//   - root mount is a tmpfs mount filled with the archive.
+	//
+	// - new PID namespace
+	//   - archive/init actually runs as PID 1.
+
+	// Whatever setup we do shouldn't affect the caller.
+	if err := syscall.Unshare(syscall.CLONE_NEWNS); err != nil {
```

Comment:
The unshare syscall is broken in Go. The syscall applies to a single kernel-level thread. Go-level threads are not forever locked to that kernel-level thread.

See: https://github.com/golang/go/issues/22283

Reply:
yes you can never use the Unshare system call. I need to push a CL to remove it from the library. YOu need to restart yourself with cmd.Unshare true. I fixed that code so it even works on systems with stupidd -- er, systemd.

Reply:
Ugh. Right. Well, they'd never remove it with the compatibility guarantee...